File: Java_ChildrenCompetitionSockets/CompetitionPersistence/src/main/java/competition/persistence/IParticipantRepository.java
package competition.persistence;


//public interface IParticipantRepository<ID, T extends HasId<ID>> extends IRepository<ID, T>{
//    T findByUsername(String username);
//    Iterable<T> findAllParticipantsForTest(int id);
//}

import competition.model.Participant;

public interface IParticipantRepository extends IRepository<Integer, Participant> {
    Participant findByUsername(String username);
    Iterable<Participant> findAllParticipantsForTest(int id);
}

File: CSharp_ChildrenCompetitionSockets/CSharp_ChildrenCompetitionSockets/persistence/IParticipantRepository.cs
using System.Collections.Generic;
using CSharp_ChildrenCompetitionGUI.model;

namespace CSharp_ChildrenCompetitionGUI.repository
{
    public interface IParticipantRepository : IRepository<int, Participant>
    {
        Participant findByUsername(string username);
        IEnumerable<Participant> findAllParticipantsForTest(int id);
    }
}